src/AMSlib/wf/workflow.hpp

```diff
@@ -165,6 +165,18 @@ class AMSWorkflow
     Store(num_elements, mInputs, outputs, predicate);
   }
 
+  bool updateModel()
+  {
+    if (!DB) return false;
+    bool local = DB->updateModel();
+#ifdef __ENABLE_MPI__
+    bool global = false;
+    MPI_Allreduce(&local, &global, 1, MPI_CXX_BOOL, MPI_LAND, comm);
+    return global;
+#else
+    return local;
+#endif
+  }
```

Comment:
This is a global sync right? Happening in every timestep?

Reply:
Yes. We could implement some relaxations:

1. We could check globally **only** if rank 0 has received something (here if `local` is true)
2. We could check every X timesteps